### PR TITLE
docs: orderbook specs

### PR DESCRIPTION
**Pull request by cosmic-vagabond**
### What has Changed?

Orderbook module specifications

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

File: x/orderbook/spec/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+### Permissionless Execution
+
+- **Open Participation:** Any address can participate in submitting orders for execution. The process is designed to be permissionless, similar to the liquidation system.
+
+### Anti-Spam Measures
+
+- **Penalties for Invalid Submissions:**
+
+  - Participants must deposit a minimum amount of Elys tokens along with their transaction.
+  - If any order in the submitted list does not meet the execution criteria, the deposit is forfeited.
```
[the remark this answers is not in the document]
Comment (cosmic-vagabond) [reply x/orderbook/spec/README.md:72]: yep good call out I will add a margin error as gov parameter the same way we did with liquidation system

File: x/orderbook/spec/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+### Off-Chain Execution Agents (Bots)
+
+- **Order Execution:** Off-chain agents will be responsible for submitting a list of order IDs that need to be executed. This eliminates the need for the blockchain to traverse all orders at every block, significantly reducing computational load.
+- **Chain Verification:** Despite offloading execution logic, the blockchain will still verify that the execution criteria of submitted orders are met before executing them.
+
+### Permissionless Execution
+
+- **Open Participation:** Any address can participate in submitting orders for execution. The process is designed to be permissionless, similar to the liquidation system.
+
+### Anti-Spam Measures
```
[the remark this answers is not in the document]
Comment (cosmic-vagabond) [reply x/orderbook/spec/README.md]: fixed

File: x/orderbook/spec/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cosmic-vagabond) [reply x/orderbook/spec/README.md:69]: margin error section added to the spec as well @amityadav0

[the remark this answers is not in the document]
Comment (amityadav0) [reply x/orderbook/spec/README.md:69]: thanks

**Review by amityadav0 — APPROVED** [review]

**Pull request merged**